### PR TITLE
Feature/hocs 2527 update with case creator changes

### PR DESCRIPTION
**Pull request by philbarton**
Add case creator to docker-compose

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+.env
+.idea
```
Comment (zuzak) [inline .gitignore:2]:
```suggestion
.idea
```

Developer-specific code should be defined in a [global gitignore file](https://sebastiandedeyne.com/setting-up-a-global-gitignore-file/) instead of at the repository level.

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -5,62 +5,62 @@ services:
   postgres:
     image: postgres:9.6
     healthcheck:
-      test: ["CMD-SHELL", "pg_isready -U postgres"]
+      test: ["CMD-SHELL", "pg_isready -U root" ]
     restart: always
     ports:
-    - 5432:5432
+      - 5432:5432
```
Comment (zuzak) [inline docker/docker-compose.yml:11]: Please try to keep unrelated whitespace changes in their own separate commit. It makes it hard to understand code reviews, and breaks `git annotate`.

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -212,17 +213,22 @@ services:
         condition: service_healthy # this waits until healthcheck passes
       aws_cli:
         condition: service_started # this doesn't
+      casework:
+        condition: service_healthy
+      documents:
+        condition: service_healthy
+      info:
+        condition: service_healthy
```
Comment (zuzak) [inline docker/docker-compose.yml:221]: Does workflow really depend on all these? (Or rather, do we really need to wait for all of these to be up before we can bring workflow up?)

File: docker/docker-compose.yml [hunk 4]
```diff
@@ -83,55 +83,58 @@ services:
...
+        aws --endpoint-url=http://localstack:4575 sns set-subscription-attributes --subscription-arn $$(aws --endpoint-url=http://localstack:4575 sns list-subscriptions-by-topic --topic-arn arn:aws:sns:eu-west-2:123456789012:hocs-audit-topic --output json | jq --raw-output '.Subscriptions[0].SubscriptionArn') --attribute-name RawMessageDelivery --attribute-value true
+        aws --endpoint-url=http://localstack:4575 sns set-subscription-attributes --subscription-arn $$(aws --endpoint-url=http://localstack:4575 sns list-subscriptions-by-topic --topic-arn arn:aws:sns:eu-west-2:123456789012:hocs-audit-topic --output json | jq --raw-output '.Subscriptions[1].SubscriptionArn') --attribute-name RawMessageDelivery --attribute-value true
+        aws --endpoint-url=http://localstack:4575 sns list-subscriptions
+        aws --endpoint-url=http://localstack:4576 sqs create-queue --queue-name ukvi-complaint-queue
+        aws --endpoint-url=http://localstack:4576 sqs create-queue --queue-name ukvi-complaint-queue-dlq
+        sleep 8h
```
Comment (zuzak) [inline docker/docker-compose.yml:119]: yikes. If it makes you feel better init containers landed in docker-compose trunk a few days ago

**Review by zuzak — COMMENTED** [review]
Please add documentation about the environment variables to docker/README.md.

Try to avoid mixing whitespace changes and "meaningful" changes as it makes reviewing this PR more difficult than it needs to be -- it's hard to figure out what's actually changed against `aws_cli`, for example.

File: docker/docker-compose.yml (hunk 2 above)
Comment (philbarton) [reply docker/docker-compose.yml:11]: I have a habit of reformatting. We should have a shared intelij format. I've added a IDEAS ticket to look at it.

File: .gitignore (hunk 1 above)
Comment (philbarton) [reply .gitignore:2]: This would cut across all projects. Added to an IDEAS ticket to discuss https://collaboration.homeoffice.gov.uk/jira/browse/HOCS-2742

File: docker/docker-compose.yml (hunk 3 above)
Comment (philbarton) [reply docker/docker-compose.yml:221]: workflow is a client of all those, workflow is flaky on start up. its less flaky with these.
It could do with more investigation.

File: docker/docker-compose.yml (hunk 4 above)
Comment (philbarton) [reply docker/docker-compose.yml:119]: I can't find any docs for that. I'll look again at the next change.

**Comment by philbarton** [comment]
Re: documenting variables. For case-creator its in that projects readme. Not sure about existing projects.
Adding them here as well would duplicate?

File: docker/docker-compose.yml (hunk 2 above)
Comment (zuzak) [reply docker/docker-compose.yml:11]: I'm not saying don't reformat; I'm merely saying be disciplined when doing so. Use `git add -p` to ensure that your commits only do one thing at a time.

**Comment by zuzak** [comment]
Rebasing the branch to split out the whitespace

**Review by zuzak — APPROVED** [review]
Reverted my force-push because you can't easily amend a commit two commits deep, and I screwed it up.

There's nothing wrong with this at a code level, but the Git commit is a bit of a mess. Commits should be atomic, and only do one thing -- these two commits do multiple things, and using the Git history to figure out what's happened in _n_ months time is going to be a headache.

I'll approve this PR, but in future please use the chunk functionality in Git to break unrelated changes into multiple commits; especially with formatting changes -- it makes it much harder to cut through the whitespace changes to figure out what the "real" changes are. Look into using `git add -p` or your IDE's equivalent next time.
